src/catalog.js

```diff
@@ -286,8 +286,10 @@ angular.module('gettext').factory('gettextCatalog', function (gettextPlurals, ge
                 cache: catalog.cache
             }).then(function (response) {
                 var data = response.data;
-                for (var lang in data) {
-                    catalog.setStrings(lang, data[lang]);
+                if (typeof data === 'object') {
```

Comment:
I hear what you are saying, but in this case the response would be an object and the status would be a 200 too. See the screenshot below.
I am willing to put in more work into this but I need some guidance from you guys.
![screen shot 2017-03-10 at 2 18 06 pm](https://cloud.githubusercontent.com/assets/358137/23809693/aa6bf78a-059c-11e7-8f2b-dcfa57d20ae4.png)